src/tiled/abstractobjecttool.cpp

```diff
@@ -454,6 +523,15 @@ void AbstractObjectTool::showContextMenu(MapObject *clickedObject,
     QAction *removeAction = menu.addAction(tr("Remove %n Object(s)", "", selectedObjects.size()),
                                            this, SLOT(removeObjects()));
 
+    // Allow the currently selected collision mask to be applied to all selected tiles in the tileset editor
+    if (auto document = DocumentManager::instance()->currentDocument()) {
+        if (auto tilesetDocument = qobject_cast<TilesetDocument*>(document)) {
+            if (tilesetDocument->selectedTiles().count() > 1) {
+                menu.addAction(tr("Apply Collision(s) to Selection"), this, SLOT(applyCollisionsToSelection()));
```

Comment:
This is a very hidden location in the UI and somewhat of a strange place also code-wise. Of course I understand the latter naturally followed from the wish to put this action in the context menu.

I think I'd rather see this action as a button alongside this magic wand button you've added to the tile collision editor. There it is easier to discover and we don't need this somewhat hacky check to see if the user is editing a tileset. Of course, it will be a nice challenge to consider a good icon and probably we need to give it an elaborate tool tip to explain the functionality.

Reply:
I decided to keep the actions in the context menu, because of the challenges of making them part of the tool bar, including finding a good icon as well as making sure the user understands that only selected objects will be copied.